core/sql/sqlcomp/nadefaults.cpp

```diff
@@ -5914,6 +5866,73 @@ enum DefaultConstants NADefaults::validateAndInsert(const char *attrName,
 
 } // NADefaults::validateAndInsert()
 
+float NADefaults::computeNumESPsPerCore(NABoolean aggressive)
+{
+   #define DEFAULT_ESPS_PER_NODE 2   // for conservation allocation
+   #define DEFAULT_ESPS_PER_CORE 0.5 // for aggressive allocation
+
+     // Make sure the gpClusterInfo points at an NAClusterLinux object.
+     // In osim simulation mode, the pointer can point at a NAClusterNSK
+     // object, for which the method numTSEsForPOS() is not defined.
+   NAClusterInfoLinux* gpLinux = dynamic_cast<NAClusterInfoLinux*>(gpClusterInfo);
+   assert(gpLinux);		
+
+   // cores per node
+   Lng32 coresPerNode = gpClusterInfo->numberOfCpusPerSMP();
+
+   if ( aggressive ) {
+      float totalMemory = gpLinux->totalMemoryAvailable(); // per Node, in KB
```

Comment:
Allocating one ESP per 2 GB seems a bit arbitrary to me. So, rather than asking the user to set the AGGRESSIVE_ESP_ALLOCATION_PER_CORE to ON (which will eventually be used, I guess), should we just ask them to pick an appropriate value for MAX_ESPS_PER_CPU_PER_OP instead?